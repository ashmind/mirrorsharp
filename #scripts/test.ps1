Set-StrictMode -Version 2
$ErrorActionPreference = 'Stop'

Write-Output 'dotnet test'
dotnet test
if ($LastExitCode -ne 0) {
    throw "dotnet test exited with code $LastExitCode"
}

Write-Output 'npm test'
try {
    Push-Location 'WebAssets'
    npm test
    if ($LastExitCode -ne 0) {
        throw "npm test exited with code $LastExitCode"
    }
}
finally {
    Pop-Location
}