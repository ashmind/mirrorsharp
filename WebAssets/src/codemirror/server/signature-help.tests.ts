import { getTooltip, showTooltip, Tooltip } from '@codemirror/view';
import { TestDriver } from '../../testing/test-driver-jest';
import { SIGNATURES_INDEX_OF } from './signature-help.test.data';

test('signature help message shows signature help', async () => {
    const driver = await TestDriver.new({ text: '_' });

    driver.receive.signatures(SIGNATURES_INDEX_OF, { start: 0, length: 1 });
    await driver.completeBackgroundWork();

    const view = driver.getCodeMirrorView();
    const tooltips = view.state.facet(showTooltip).filter(t => t) as ReadonlyArray<Tooltip>;
    expect(tooltips.length).toBe(1);
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const tooltipView = getTooltip(view, tooltips[0]!)!;
    expect(tooltipView.dom.innerHTML).toMatchSnapshot();
});

test('signature help message with empty signatures hides signature help', async () => {
    const driver = await TestDriver.new({ text: '_' });

    driver.receive.signatures(SIGNATURES_INDEX_OF, { start: 0, length: 1 });
    await driver.completeBackgroundWork();
    driver.receive.signatures();

    const view = driver.getCodeMirrorView();
    const tooltips = view.state.facet(showTooltip).filter(t => t) as ReadonlyArray<Tooltip>;
    expect(tooltips).toEqual([]);
});

test('Ctrl+Shift+Space requests signature list', async () => {
    const driver = await TestDriver.new({ text: '' });

    driver.domEvents.keydown(' ', { ctrlKey: true, shiftKey: true });
    await driver.completeBackgroundWork();

    expect(driver.socket.sent.slice(-1)[0]).toBe('PF');
});