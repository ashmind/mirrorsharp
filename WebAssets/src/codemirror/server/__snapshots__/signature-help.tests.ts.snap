// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signature help message shows signature help 1`] = `"<li class="mirrorsharp-signature mirrorsharp-signature--selected"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">char</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">)</span><div class="mirrorsharp-signature-info"><span class="tok-text">Reports the zero-based index of the first occurrence of the specified Unicode character in this string.</span></div><div class="mirrorsharp-signature-info-parameter"><span class="mirrorsharp-signature-info-parameter-name"></span><span class="tok-text">A Unicode character to seek.</span></div></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">string</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">)</span></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">char</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">startIndex</span><span class="tok-punctuation ">)</span></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">char</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-namespace ">System</span><span class="tok-punctuation ">.</span><span class="tok-enum ">StringComparison</span><span class="tok-space "> </span><span class="tok-parameter ">comparisonType</span><span class="tok-punctuation ">)</span></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">string</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">startIndex</span><span class="tok-punctuation ">)</span></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">string</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-namespace ">System</span><span class="tok-punctuation ">.</span><span class="tok-enum ">StringComparison</span><span class="tok-space "> </span><span class="tok-parameter ">comparisonType</span><span class="tok-punctuation ">)</span></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">char</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">startIndex</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">count</span><span class="tok-punctuation ">)</span></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">string</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">startIndex</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">count</span><span class="tok-punctuation ">)</span></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">string</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">startIndex</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-namespace ">System</span><span class="tok-punctuation ">.</span><span class="tok-enum ">StringComparison</span><span class="tok-space "> </span><span class="tok-parameter ">comparisonType</span><span class="tok-punctuation ">)</span></li><li class="mirrorsharp-signature"><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-keyword ">string</span><span class="tok-punctuation ">.</span><span class="tok-method ">IndexOf</span><span class="tok-punctuation ">(</span><span class="tok-keyword mirrorsharp-signature-part--selected">string</span><span class="tok-space mirrorsharp-signature-part--selected"> </span><span class="tok-parameter mirrorsharp-signature-part--selected">value</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">startIndex</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-keyword ">int</span><span class="tok-space "> </span><span class="tok-parameter ">count</span><span class="tok-punctuation ">,</span><span class="tok-space "> </span><span class="tok-namespace ">System</span><span class="tok-punctuation ">.</span><span class="tok-enum ">StringComparison</span><span class="tok-space "> </span><span class="tok-parameter ">comparisonType</span><span class="tok-punctuation ">)</span></li>"`;
